astropy/coordinates/angles/core.py

```diff
@@ -514,21 +503,17 @@ def is_within_bounds(self, lower=None, upper=None):
         is_within_bounds : bool
             `True` if all angles satisfy ``lower <= angle < upper``
         """
-        ok = True
-        if lower is not None:
-            ok &= np.all(Angle(lower) <= self)
-        if ok and upper is not None:
-            ok &= np.all(self < Angle(upper))
-        return bool(ok)
+        # Avoid `any()` so that a NaN value would return `False`
+        if lower is not None and not (Angle(lower) <= self).all():
```

Comment:
Maybe combine all?
```
return (lower is None or np.all(self >= Angle(lower))) and (upper is None or np.all(self < Angle(upper)))
```

Reply:
I wrote the comment to warn against replacing `not (Angle(lower) <= self).all()` with `(Angle(lower) > self).any()`, but with your suggestion the comment is not needed anymore.